### ISSUE TITLE
very slow startup

### ISSUE THREAD

**Issue by goertzenator** [goertzenator/rust.mk#4]
`rust.mk` installs the `cargo-erlangapp` crate.  Due to JSON deps this is very slow, so consider replacing that crate with an Erlang app.

@essen, off the top of your head are you aware of an erlang.mk plugin that has nested dependencies that I could use as a reference?  It is not clear to me how `plugin.mk` rules should invoke functions in the plugin app while keeping dependent apps accessible.

**Comment by essen** [comment]
Your plugin should be able to `DEPS += whatyouneed`, if that's what you mean?

**Comment by goertzenator** [comment]
That's right.  I'm looking for an example of a plugin that does exactly that and uses those deps at build time.

**Comment by essen** [comment]
I don't think I currently *know* of one, but I wanted to *write* one for a project.

**Comment by goertzenator** [comment]
Okay, no worries then.  I'll try to figure things out.

**Comment by essen** [comment]
FYI RabbitMQ has a lot of plugins doing this sort of thing. For example: https://github.com/rabbitmq/rabbitmq-common/blob/master/mk/rabbitmq-early-test.mk#L92
